### PR TITLE
feat(mdx-table): new styles

### PR DESCRIPTION
**Pull request by thiskevinwang**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](https://app.asana.com/0/1202097197789424/1202954349649041/f) 🎟️
- [Figma](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=7288%3A58665)

Related:
- https://github.com/hashicorp/dev-portal/pull/940

## 🗒️ What

New MDX table styles

CleanShot 2022-09-15 at 15 08 01@2x

## 📸 Design Screenshots

| Before | After |
| ---    | ---   |
| CleanShot 2022-09-14 at 15 49 15@2x | CleanShot 2022-09-15 at 15 09 09@2x |

## 🧪 Testing

Visit the following pages for some live usages:
- https://dev-portal-git-kevin-table-styles-hashicorp.vercel.app/consul/api-docs/txn
- https://dev-portal-git-kevin-table-styles-hashicorp.vercel.app/consul/api-docs/status
- https://dev-portal-git-kevin-table-styles-hashicorp.vercel.app/consul/api-docs/admin-partitions
- https://dev-portal-git-kevin-table-styles-hashicorp.vercel.app/consul/docs/agent/telemetry

GitHub search — for grok'ing the MDX implementation 
- https://cs.github.com/hashicorp/consul?q=%3A--

### CODE REVIEW

File: src/components/dev-dot-content/mdx-components/mdx-table/mdx-table.module.css [hunk 1]
```diff
@@ -3,72 +3,59 @@ We are using a wrapper here to enable horizontal scrolling while still retaining
...
-      border-top: 1px solid #ddd;
-    }
-
-    & table {
-      background-color: #fff;
-    }
-
-    & tr.warning td {
-      background: #fcf8e3;
-    }
```
Comment (thiskevinwang) [inline src/components/dev-dot-content/mdx-components/mdx-table/mdx-table.module.css:72]: 💭 Curious how to reproduce the `.warning` class

Comment (thiskevinwang) [reply src/components/dev-dot-content/mdx-components/mdx-table/mdx-table.module.css:72]: I'm thinking that this is not actually used.
`.warning` would result in a CSS-module-scoped className (like "mdx_table_warning__ag24q"), but nothing in the react component actually specifies `className={s.warning}`

**Review by alexcarpenter — COMMENTED** [review]
Looks like we're missing the `overflow-x: auto` CSS to enable horizontal scrolling on mobile when needed.

CleanShot 2022-09-14 at 17 04 20@2x

File: src/components/dev-dot-content/mdx-components/mdx-table/mdx-table.module.css [hunk 2]
```diff
@@ -3,72 +3,61 @@ We are using a wrapper here to enable horizontal scrolling while still retaining
...
-  }
+.tableWrapper {
```
Comment (alexcarpenter) [inline src/components/dev-dot-content/mdx-components/mdx-table/mdx-table.module.css:7]: Thoughts on adding the border to the wrapper to ensure the outer border is always visible, even when the table scrolls? On the right you can see that the wrapping border is visible vs the border being hidden?

CleanShot 2022-09-15 at 09 53 39@2x

Accompanying CSS:

```css
.tableWrapper {
	--background-color: #fafafa;
	--border-color: rgba(101, 106, 118, 0.2);
	--border-radius: 6px;

	border: 1px solid var(--border-color);
	overflow: hidden;
	border-radius: var(--border-radius);
	width: fit-content;
	max-width: 100%;
	overflow-x: auto;

	& table {
		border-collapse: collapse;
		border-style: hidden;
	}

	& thead {
		background-color: var(--background-color);
	}

	& tbody tr:nth-child(even) {
		background-color: var(--background-color);
	}

	& th,
	& td {
		border: 1px solid var(--border-color);
		padding: 8px 12px;
	}
}

```

Is there accompanying HDS variables we should be using for the border and background colors here?

Comment (braydenlove) [reply src/components/dev-dot-content/mdx-components/mdx-table/mdx-table.module.css:7]: I think I prefer the one on the left.

Comment (braydenlove) [reply src/components/dev-dot-content/mdx-components/mdx-table/mdx-table.module.css:7]: I think I prefer the one on the left.

**Review by braydenlove — COMMENTED** [review]
Hey @thiskevinwang thanks for jumping on this. We want to go full width on all the tables tho to avoid inconsistent widths. Let us know what you think.

**Review by alexcarpenter — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**